apps/portals/nf/src/pages/NFSearchPage.tsx

```diff
@@ -0,0 +1,193 @@
+export const searchPageTabs: PortalSearchTabConfig[] = [
+  {
+    title: 'Initiatives',
+    path: 'Initiatives',
+  },
+  {
+    title: 'Studies',
+    path: 'Studies',
+  },
+  {
+    title: 'Datasets',
+    path: 'Datasets',
+  },
+  {
+    title: 'Publications',
+    path: 'Publications',
+  },
+  {
+    title: 'Tools',
+    path: 'Tools',
+  },
+  {
+    title: 'Hackathons',
+    path: 'Hackathons',
+  },
+]
+
+export const searchPageChildRoutes: RouteObject[] = [
+  {
+    index: true,
+    element: <NFSearchPage />,
+  },
+  {
+    path: searchPageTabs[0].path,
+    element: <NFSearchPage selectedTabIndex={0} />,
+  },
+  {
+    path: searchPageTabs[1].path,
+    element: <NFSearchPage selectedTabIndex={1} />,
+  },
+  {
+    path: searchPageTabs[2].path,
+    element: <NFSearchPage selectedTabIndex={2} />,
+  },
+  {
+    path: searchPageTabs[3].path,
+    element: <NFSearchPage selectedTabIndex={3} />,
+  },
+  {
+    path: searchPageTabs[4].path,
+    element: <NFSearchPage selectedTabIndex={4} />,
+  },
+  {
+    path: searchPageTabs[5].path,
+    element: <NFSearchPage selectedTabIndex={5} />,
+  },
+]
+
+export type NFSearchPageProps = {
+  selectedTabIndex?: number
+}
+
+function getQueryCount(queryResultBundleJSON: string) {
+  const queryResultBundle = JSON.parse(
+    queryResultBundleJSON,
+  ) as QueryResultBundle
+  const { queryCount } = queryResultBundle
+  return queryCount
+}
+
+export function NFSearchPage(props: NFSearchPageProps) {
+  const { selectedTabIndex } = props
+  // Note: Files does not currently enable FTS
+  const { initiatives, studies, datasets, publications, tools, hackathons } =
+    nfConfigs
+  const [searchPageTabsState, setSearchPageTabsState] =
+    useState<PortalSearchTabConfig[]>(searchPageTabs)
+  const navigate = useNavigate()
+  const location = useLocation()
+
+  const onQueryResultBundleChange = useCallback(
+    (tabIndex: number, newQueryResultBundleJSON: string) => {
+      const newCount = getQueryCount(newQueryResultBundleJSON)
+      if (searchPageTabsState[tabIndex].count !== newCount) {
+        searchPageTabs[tabIndex].count = newCount
+        setSearchPageTabsState([...searchPageTabs])
+      }
+      // PORTALS-3382: If no tab is selected and all counts have been set, then redirect to the item with the highest count
+      const allCountsSet = searchPageTabs.every(tab => tab.count !== undefined)
+      if (selectedTabIndex == undefined && allCountsSet) {
+        // Navigate to the tab that has the highest count.
+        // Explicitly initialize the accumulator ("max") to the first element (searchPageTabs[0]). "max" will never be null
+        const maxCountTab = searchPageTabs.reduce(
+          (max, tab) => (tab.count! > max.count! ? tab : max),
+          searchPageTabs[0],
+        )
+        navigate({
+          pathname: `/Search/${maxCountTab.path}`,
+          search: location.search,
+        })
+      }
+    },
+    [searchPageTabsState, selectedTabIndex],
```

Comment:
can this logic shared between CCKP and NF be parameterized, extracted, and shared?